### PR TITLE
feat: prevent only user triggered events when business process

### PR DESCRIPTION
**Pull request by dharmendrak**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/unspec/callback/CaseEvent.java [hunk 1]
```diff
@@ -6,25 +6,33 @@
...
-    NOTIFY_RESPONDENT_SOLICITOR1_FOR_CLAIM_ISSUE("Notify claim issue"),
-    NOTIFY_APPLICANT_SOLICITOR1_FOR_REQUEST_FOR_EXTENSION("Notify request for extension"),
-    NOTIFY_RESPONDENT_SOLICITOR1_FOR_EXTENSION_RESPONSE("Notify extension response"),
-    NOTIFY_APPLICANT_SOLICITOR1_FOR_SERVICE_ACKNOWLEDGEMENT("Notify service acknowledgement"),
-    NOTIFY_APPLICANT_SOLICITOR1_FOR_DEFENDANT_RESPONSE("Notify response"),
-    NOTIFY_RESPONDENT_SOLICITOR1_FOR_CASE_HANDED_OFFLINE("Notify handed offline"),
-    NOTIFY_APPLICANT_SOLICITOR1_FOR_CASE_HANDED_OFFLINE("Notify handed offline"),
-    NOTIFY_RESPONDENT_SOLICITOR1_FOR_CASE_TRANSFERRED_TO_LOCAL_COURT("Notify transferred local court"),
-    NOTIFY_APPLICANT_SOLICITOR1_FOR_CASE_TRANSFERRED_TO_LOCAL_COURT("Notify transferred local court");
+    CREATE_CLAIM("Create claim", UserType.USER),
```
Comment (pd-dudkiewicz) [inline src/main/java/uk/gov/hmcts/reform/unspec/callback/CaseEvent.java]: can we have static imports for user types?

File: src/test/java/uk/gov/hmcts/reform/unspec/aspect/NoOngoingBusinessProcessAspectTest.java [hunk 2]
```diff
@@ -43,76 +45,132 @@
...
-                                 .businessProcess(BusinessProcess.builder().status(status).build())
-                                 .build()).build())
-                         .build())
-            .build();
-        Object result = aspect.checkOngoingBusinessProcess(proceedingJoinPoint, callbackParams);
-
-        assertThat(result).isEqualTo(response);
-        verify(proceedingJoinPoint).proceed();
+    @Nested
+    class NormalUserEvent {
```
Comment (sabahirfan) [inline src/test/java/uk/gov/hmcts/reform/unspec/aspect/NoOngoingBusinessProcessAspectTest.java]: I will think to rename this.

**Review by sabahirfan — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/reform/unspec/aspect/NoOngoingBusinessProcessAspectTest.java (hunk 2 above)
Comment (swalker125) [reply src/test/java/uk/gov/hmcts/reform/unspec/aspect/NoOngoingBusinessProcessAspectTest.java]: I would go with UserEvent and CamundaEvent

File: src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOngoingBusinessProcessAspect.java [hunk 3]
```diff
@@ -37,7 +38,7 @@ public Object checkOngoingBusinessProcess(
         CaseEvent caseEvent = CaseEvent.valueOf(callbackParams.getRequest().getEventId());
         CaseDetails caseDetails = callbackParams.getRequest().getCaseDetails();
         CaseData caseData = caseDetailsConverter.toCaseData(caseDetails);
-        if (hasNoOngoingBusinessProcess(caseData)) {
+        if (hasNoOngoingBusinessProcess(caseData) || caseEvent.getUserType() == UserType.CAMUNDA) {
```
Comment (swalker125) [inline src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOngoingBusinessProcessAspect.java]: could we wrap this logic in caseEvent? something like caseEvent.isCamundaEvent()

Comment (swalker125) [reply src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOngoingBusinessProcessAspect.java]: @dharmendrak thoughts on this?

**Comment by swalker125** [comment]
would also rename this PR title to something like

feat: prevent only user triggered events when business process

**Review by sabahirfan — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOngoingBusinessProcessAspect.java (hunk 3 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/reform/unspec/aspect/NoOngoingBusinessProcessAspect.java]: @swalker125  done

**Review by swalker125 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/unspec/callback/CaseEvent.java [hunk 4]
```diff
@@ -3,28 +3,43 @@
...
+    NOTIFY_APPLICANT_SOLICITOR1_FOR_CASE_TRANSFERRED_TO_LOCAL_COURT(
+        "Notify transferred local court",
+        CAMUNDA
+    ),
+    START_BUSINESS_PROCESS("Start business process", CAMUNDA);
 
     private final String displayName;
+    private final UserType userType;
+
+    public boolean isCamundaEvent() {
```
Comment (swalker125) [inline src/main/java/uk/gov/hmcts/reform/unspec/callback/CaseEvent.java:43]: hmm I know simple but let's have a unit test too

Comment (swalker125) [reply src/main/java/uk/gov/hmcts/reform/unspec/callback/CaseEvent.java:43]: actually ignore me, this is fine

**Review by swalker125 — DISMISSED** [review]

**Review by pd-dudkiewicz — DISMISSED** [review]

**Review by pd-dudkiewicz — APPROVED** [review]

**Review by sabahirfan — APPROVED** [review]

**Pull request merged**